### PR TITLE
Modularize the lang-mustache component

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This change modularizes the lang-mustache component, by adding a module-info.java

relates #78744

### CODE REVIEW

File: modules/lang-mustache/src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ */
+
+module org.elasticsearch.mustache {
+    requires org.elasticsearch.base;
+    requires org.elasticsearch.server;
+    requires org.elasticsearch.xcontent;
+    requires org.apache.logging.log4j;
+    requires com.github.mustachejava;
+
+    exports org.elasticsearch.script.mustache;
```
Comment (rjernst) [inline modules/lang-mustache/src/main/java/module-info.java:16]: Where is the export needed? Is this only needed by server for reflective construction?

**Review by rjernst — APPROVED** [review]
Looks good, one minor question.

File: modules/lang-mustache/src/main/java/module-info.java (hunk 1 above)
Comment (ChrisHegarty) [reply modules/lang-mustache/src/main/java/module-info.java:16]: The HLRC uses types from `org.elasticsearch.script.mustache`, see `org.elasticsearch.client.RequestConverters`.

Comment (rjernst) [reply modules/lang-mustache/src/main/java/module-info.java:16]: Ok, perhaps add a TODO to remove this once HLRC is removed (it's dead code, just used by tests now)

**Pull request merged**